streams/src/main/java/org/apache/kafka/streams/state/internals/metrics/Sensors.java

```diff
@@ -44,5 +51,67 @@ public static Sensor createTaskAndStoreLatencyAndThroughputSensors(final Sensor.
         addInvocationRateAndCount(sensor, metricsGroup, storeTags, operation);
         return sensor;
     }
+
+    public static Sensor createBufferSizeSensor(final StateStore store,
+                                                final InternalProcessorContext context) {
+        return getBufferSizeOrCountSensor(store, context, "size");
+    }
+
+    public static Sensor createBufferCountSensor(final StateStore store,
+                                                 final InternalProcessorContext context) {
+        return getBufferSizeOrCountSensor(store, context, "count");
+    }
+
+    private static Sensor getBufferSizeOrCountSensor(final StateStore store,
+                                                     final InternalProcessorContext context,
+                                                     final String property) {
+        final StreamsMetricsImpl metrics = context.metrics();
+
+        final String sensorName = "suppression-buffer-" + property;
+
+        final Sensor sensor = metrics.storeLevelSensor(
+            context.taskId().toString(),
+            store.name(),
+            sensorName,
+            Sensor.RecordingLevel.DEBUG
+        );
+
+        final String metricsGroup = "stream-buffer-metrics";
```

Comment:
In the KIP, I erroneously proposed to make this a node-level sensor. It should be a store-level sensor instead. I'm proposing to update the KIP if you all agree.

Reply:
Yeah, that makes sense to me. +1

Reply:
This is not newly introduced in this PR but: although we call it `stream-buffer-metrics` where the `buffer-id` is the store-name, BUT the store name could be an auto-generated name. Note for the named cache we call it `stream-record-cache-metrics` but the cache name is the store name, and hence we have the same issue.

But here we use the store.name which would be `KTABLE-SUPPRESS-XXXX-store` if it is auto generated, maybe it's less confusing to use the processor-name instead which would be `KTABLE-SUPPRESS-XXX` if it is auto generated. Note even if `Suppressed.name()` is specified as `XYZ` with store.name() we still set the buffer-id as `XYZ-store` instead.

Reply:
I think it's good to be aware of this, but it seems like reporting the store name is the right thing to do here.

Note that the buffer in this case *is* the store for the suppression processor. So, using the processor name instead of the full store name would be more analogous to doing the same thing for a K/V Store.

As written, the metrics that are specific to the buffer (aka store) will be tagged with a name that matches the name of that component when you describe the topology.

Does that seem right, or have I missed your point?